pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTaskRestletResource.java

```diff
@@ -419,12 +440,17 @@ public SuccessResponse toggleTaskQueueState(
   @DELETE
   @Path("/tasks/{taskType}")
   @Authenticate(AccessType.DELETE)
-  @ApiOperation("Delete all tasks (as well as the task queue) for the given task type")
+  @ApiOperation("Delete specified tasks or all the tasks (as well as the task queue) for the given task type")
```

Comment:
I'd suggest adding a separate API for deleting task `DELETE /tasks/task`.
Do we need to support deleting multiple tasks, or always one at a time? Deleting multiple tasks might fail in the middle and left half of the tasks.
IIRC, in order to delete a task, we need to first stop the task queue?

Reply:
correct, deleting a job requires to stop the task queue firstly; otherwise, the method throws exception. 

yeah, I can start with deleting one task a time for simplicity, and that'd be enough for most use cases today, because many of our minion tasks stop scheduling new tasks if there is one still running.